src/overlay/utils/position.js

```diff
@@ -110,10 +110,10 @@ export default class Position {
             }
         }
 
+        // This will only execute if `pinElement` could not be placed entirely in the Viewport
         const inViewportLeft = this._makeElementInViewport(pinElement, firstPositionResult.left, 'Left', isPinFixed);
         const inViewportTop = this._makeElementInViewport(pinElement, firstPositionResult.top, 'Top', isPinFixed);
-
-        this._setPinElementPostion(pinElement, {left: inViewportLeft, top: inViewportTop});
+        this._setPinElementPostion(pinElement, { left: inViewportLeft, right: inViewportTop }, this.offset);
```

Comment:
So this is the main change?

Reply:
yes. `this.offset` will now always be added to the position